File: spring_db_practice/src/main/java/com/yunhalee/spring_db_practice/repository/order/OrderRepository.java
package com.yunhalee.spring_db_practice.repository.order;

import com.yunhalee.spring_db_practice.domain.Order;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

@Repository
public interface OrderRepository extends JpaRepository<Order, Long> {
}

File: msa_architecture/msa_architecture_service/order_service/order-domain/order-application-service/src/main/kotlin/com/yunhalee/msa/architecture/order/domain/application/service/domain/ports/output/repository/OrderRepository.kt
package com.yunhalee.msa.architecture.order.domain.application.service.domain.ports.output.repository

import com.yunhalee.msa.architecture.common.domain.valueobject.OrderId
import com.yunhalee.msa.architecture.service.order.domain.core.entity.Order
import com.yunhalee.msa.architecture.service.order.domain.core.valueobject.TrackingId
import java.util.Optional

interface OrderRepository {
    fun save(order: Order): Order

    fun findById(orderId: OrderId): Optional<Order>

    fun findByTrackingId(trackingId: TrackingId): Optional<Order>

}